### PR TITLE
[Buttons] Implement a semantic color scheme color themer API.

### PR DESCRIPTION
**Pull request by yarneo**
Addition of a unit test, and updated example to use the new themer.

Pivotal: https://www.pivotaltracker.com/story/show/156169780

Flat button and raised button theming are separate stories and will be added in separate PRs.

simulator screen shot - iphone 8 - 2018-04-10 at 15 09 20

### CODE REVIEW

File: components/Buttons/src/ColorThemer/MDCButtonColorThemer.h [hunk 1]
```diff
@@ -15,17 +15,27 @@
...
+ @param colorScheme The color scheme to apply to MDCButton.
+ */
++ (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
+                        toButton:(nonnull MDCButton *)button;
+
 /**
  Applies a color scheme to theme a MDCButton. Use a UIAppearance proxy to apply a color scheme to
  all instances of MDCButton.
 
+ This method is deprecated. Consider using applySemanticColorScheme:colorScheme.
```
Comment (jverkoey) [inline components/Buttons/src/ColorThemer/MDCButtonColorThemer.h]: “Will soon be”

**Review by jverkoey — APPROVED** [review]
Lgtm.

**Comment by yarneo** [comment]
Hey @jverkoey I had noticed something I hadn't prior but for MDCButton we have a `disabledAlpha` in place that is defaulted to 0.12f. If we were to leave it as is it would apply a `self.alpha` on the entire UIButton and its subviews (including the title).

However the title needs to have a higher alpha which would cause an issue and we would need to set the `disabledAlpha` to 0.26f and then set the alpha of the backroundColor to 0.12f/0.26f which will be too complicated. Therefore I have set it to 1.f in the themer. Want to make sure if this approach looks good for you.

**Comment by jverkoey** [comment]
Given the way that property is used I agree that you'll have to set it to 1.f in the themer. LGTM.

**Pull request merged**